### ISSUE TITLE
Upgrade to Tycho version 4.0.5 to address proxy issues

### ISSUE THREAD

**Issue by baronswindle** [microsoft/vscode-java-test#1656]
Tycho v4.0.3 and v4.0.4 — and possibly earlier versions as well — do not seem to handle proxy settings very well. Like many big companies, my employer only allows access to the internet via a proxy, and when I attempted to run `npm run build-plugin` in this repository, I got the following error message.

```log
Failed to resolve target definition file:/path/to/vscode-java-test/java-extension/com.microsoft.java.test.target/com.microsoft.java.test.tp.target: Failed to load p2 metadata repository from location https://download.eclipse.org/buildship/updates/e423/snapshots/3.x/3.1.7.v20221108-1729-s/: Unable to read repository at https://download.eclipse.org/buildship/updates/e423/snapshots/3.x/3.1.7.v20221108-1729-s. Remote host terminated the handshake: Connection reset
```

Tycho v4.0.5 includes [this change](https://github.com/eclipse-tycho/tycho/pull/3303), which appears to address the issue. See [change log for v4.0.5](https://github.com/eclipse-tycho/tycho/releases/tag/tycho-4.0.5). And when I updated `tycho-version` from 4.0.3 to 4.0.5 locally, `npm run build-plugin` completely succeeded for me.

Are y'all open to this change?

**Comment by jdneo** [comment]
Yes, it's totally fine to upgrade it if it can solve the issue.

PR is welcome.

**Comment by jdneo** [comment]
addressed by #1664
